dstep/translator/Translator.d

```diff
@@ -69,48 +70,62 @@ class Translator
         write(outputFile, translateToString());
     }
 
+    static size_t countTrailingEndls(string a) 
+    {
+        size_t itr = a.length;
+
+        while (itr != 0 && a[itr - 1] == '\n')
+            --itr;
+                
+        return a.length - itr;
+    }
+
+    static size_t countLeadingEndls(string a)
+    {
+        size_t count = 0;
+
+        while (count < a.length && a[count] == '\n')
+            ++count;
+
+        return count;
+    }
+
+    static void endlAwareAppend(ref Appender!string a, string b)
+    {
+        import std.algorithm.comparison;
+
+        size_t aTrailing = countTrailingEndls(a.data);
+        size_t bLeading = countLeadingEndls(b);
+        size_t bTrailing = countTrailingEndls(b);
+
+        size_t num = min(bLeading + aTrailing, 2);
+        a.put("\n".replicate(num < aTrailing ? 0 : num - aTrailing));
+        a.put(b[bLeading..$-bTrailing]);
+        a.put("\n".replicate(min(bTrailing, 2)));
+    }
```

Comment:
Seems reasonable, but I see some issues.

> each node in the AST
So we would need to have some dstep AST (currently there are strings only) or use the clang AST but there may be no one to one mapping between dstep AST and clang AST.

> Add a newline after each node
What if there is a few nodes in one line?

> I also have information if there's a node coming after the current node at the same level (a next node).
I suppose it is the easiest to do if a full AST is known in advance.